packages/components/number-field/src/number-field.scss

```diff
@@ -9,34 +9,4 @@
 .step-buttons {
   display: flex;
   margin-inline-end: calc(var(--sl-size-025) - var(--sl-size-borderWidth-subtle));
-
-  button,
-  ::slotted(button) {
-    block-size: calc(1lh + (var(--sl-size-050) + var(--sl-size-borderWidth-subtle)) * 2);
-    inline-size: calc(1lh + (var(--sl-size-050) * 2));
-
-    sl-icon,
-    ::slotted(sl-icon) {
-      --sl-icon-container-size: calc(var(--sl-size-icon-md) + (var(--sl-size-050) * 2));
-
-      inline-size: calc(var(--sl-size-icon-md) + (var(--sl-size-050) * 2));
-    }
-  }
-}
-
-:host([size='lg']) .step-buttons {
-  button,
-  ::slotted(button) {
-    block-size: calc(1lh + (var(--sl-size-100) + var(--sl-size-borderWidth-subtle)) * 2);
-    inline-size: calc(1lh + (var(--sl-size-100) * 2));
```

Comment:
In that version for the `step buttons` at the `end` it was intentional before to have `inline-size` a little bit smaller than the `block-size` in that case. @arecuenco-dsgn can you please also take a look?